### PR TITLE
Use lates ocm library and some refactor

### PR DESCRIPTION
**Pull request by qiujian16**

### CODE REVIEW

**Comment by tamalsaha** [comment]
This should also fix the missing patch permission https://github.com/open-cluster-management-io/ocm/pull/190

File: pkg/controllers/ocmcontroller/ocmcontroller.go [hunk 1]
```diff
@@ -112,189 +99,64 @@ func runControllers(ctx context.Context,
...
-			clusterClient.ClusterV1beta2(),
-			clusterInformers.Cluster().V1beta2().ManagedClusterSets(),
-			controllerContext.EventRecorder,
-		)
-		globalManagedClusterSetController = managedclusterset.NewGlobalManagedClusterSetController(
-			clusterClient.ClusterV1beta2(),
-			clusterInformers.Cluster().V1beta2().ManagedClusterSets(),
-			controllerContext.EventRecorder,
+	if features.HubMutableFeatureGate.Enabled(ocmfeature.ManifestWorkReplicaSet) {
+		go workhub.RunControllerManagerWithInformers(
```
Comment (skeeey) [inline pkg/controllers/ocmcontroller/ocmcontroller.go:133]: This also will enable the work hub controller (ManifestWorkReplicaSet) 
/cc @tamalsaha

**Comment by skeeey** [comment]
@qiujian16  there are some unused and errcheck in the verify steps
https://github.com/open-cluster-management-io/multicluster-controlplane/actions/runs/5935720333/job/16094687054?pr=110#step:4:337

**Comment by qiujian16** [comment]
seems there is some flaky in e2e, retesting passed.

**Pull request merged**